elements/lisk-dpos/src/dpos.ts

```diff
@@ -152,6 +156,37 @@ export class Dpos {
 		return this.rounds.calcRoundStartHeight(activeRounds);
 	}
 
+	public async isActiveDelegate(
+		address: string,
+		height: number,
+	): Promise<boolean> {
+		const relevantRound = this.rounds.calcRound(height);
+		const voteWeightsStr = await this.chain.dataAccess.getConsensusState(
+			CONSENSUS_STATE_VOTE_WEIGHTS_KEY,
+		);
+		const voteWeights: VoteWeights = voteWeightsStr
+			? JSON.parse(voteWeightsStr)
+			: [];
+
+		const voteWeight = voteWeights.find(
+			roundRecord => roundRecord.round === relevantRound,
+		);
+
+		if (!voteWeight) {
+			throw new Error(
+				`Vote weight not found for round ${relevantRound} for the given height ${height}`,
+			);
+		}
+		const activeDelegateVoteWeights = voteWeight.delegates.slice(
+			0,
+			this._activeDelegates,
+		);
+
+		return (
+			activeDelegateVoteWeights.findIndex(vw => vw.address === address) > -1
+		);
```

Comment:
Why the logic of `isActiveDelegate` and `isStandbyDelegate` is different? It could be really simple, something which is in the forger list and not standby is an active delegate.

Reply:
forgerlist can be only generated right at the round, but what we need is active delegate list in the future. Therefore, it's using the voteWeights

Reply:
We can't look into the unseen future. Can we rename this function to say `isActiveDelegateInTwoFutureRounds` or something similar. Otherwise, the context of `isActiveDelegate` and `isStandbyDelgate` will always be considered the current round.

Reply:
`isActiveDelegate` and `isStandbyDelgate` are both checking as much as they can.
isActiveDelegate will check with voteWeights because it has most information, and it can check from the finalized - current round + 2, for `isStandbyDelegate` it's finalized - current round

so renaming will not describe what it is fully. Maybe ill add comments on function?